### PR TITLE
TechDocs: Update roadmap with milestones

### PR DESCRIPTION
**Pull request by OrkoHunter**
This PR

- updates the [TechDocs overview](https://backstage.io/docs/features/techdocs/techdocs-overview) page with updated milestones.
- closes the feedback form.
- addresses other small updates.

The GitHub milestones are already tagged with issues and will receive more newer issues in coming days.

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4679-umj7ggvtogwbju9pjkunyqbqbwue8qyb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60364b03f285af846f96e94c

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
(didn't read, honestly - approved to unblock)

**Review by garyniemen — APPROVED** [review]

File: docs/features/techdocs/README.md [hunk 1]
```diff
@@ -117,7 +58,47 @@ providers are used.
...
+**General Availability (GA) release** -
+[Milestone](https://github.com/backstage/backstage/milestone/30)
+
+- Bugs are rare, TechDocs APIs are stable and scales easily in large
+  organizations.
+- Better integration with
+  [Scaffolder V2](https://github.com/backstage/backstage/issues/2771) (e.g. easy
+  to choose and plug docs with Software Templates).
+- Possible to configure several aspects about TechDocs e.g. URL, homepage,
+  theme.
```
Comment (adamdmharvey) [inline docs/features/techdocs/README.md:85]: * the first bullet uses an exempli gratia in parenthesis, but the second leaves it open as part of the sentence? suggest to move the second bullet part starting with `e.g.` into parens also?
* also what does it mean to "plug docs with" ? wonder if that's missing something

**Review by adamdmharvey — COMMENTED** [review]
🎉 minor nit

File: docs/features/techdocs/README.md (hunk 1 above)
Comment (OrkoHunter) [reply docs/features/techdocs/README.md:85]: 👍 

Reworded to "plug documentation template with Software Templates". Idea is "Composing templates together (#docs-as-code) usecase" from https://github.com/backstage/backstage/issues/2771

On GitHub, when we create a new repository, we get to choose the type of .gitignore file and License type. Similarly, using Scaffolder one will create a new Library, a new Backend service, a new CLI and then choose from a list of available "Documentation templates" available in the organization. In Spotify, our technical writers have several such documentation templates available for different use-cases (which we plan to make public with the Software Templates Marketplace work https://github.com/backstage/backstage/issues/3383).

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4679-umj7ggvtogwbju9pjkunyqbqbwue8qyb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60364b03f285af846f96e94c

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4679-umj7ggvtogwbju9pjkunyqbqbwue8qyb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60364b03f285af846f96e94c

**Pull request merged**